drivers/android/context.rs

```diff
@@ -27,20 +31,15 @@ unsafe impl Sync for Context {}
 
 impl Context {
     pub(crate) fn new() -> Result<Arc<Self>> {
-        let mut ctx = Pin::from(UniqueArc::try_new(Self {
-            // SAFETY: Init is called below.
-            manager: unsafe {
-                Mutex::new(Manager {
+        let ctx = UniqueArc::pin_init::<core::convert::Infallible>(pin_init!(Self {
+            manager: mutex!(
```

Comment:
I think this should be called `new_mutex` or something else to that effect, just `mutex` is too generic. The same applies to the macros for other primitives.

Reply:
I would call it `mutex_new`.

Reply:
I suggest `XXX_new` because kernel C function always use that convention (and it sounds the same as `XXX::new`).